src/libstd/sys/unix/process/process_unix.rs

```diff
@@ -198,7 +198,11 @@ impl Command {
                 // privilege dropping function.
                 //FIXME: Redox kernel does not support setgroups yet
                 #[cfg(not(target_os = "redox"))]
-                let _ = libc::setgroups(0, ptr::null());
+                let _ = if let Some(g) = self.get_groups() {
+                    libc::setgroups(g.len(), g.as_ptr())
+                } else {
+                    libc::setgroups(0, ptr::null())
+                };
```

Comment:
This is not the right place to put this code. This code is using `setgroups` to drop supplementary groups when changing UID.

I think the logic should be something like this:

- If we were given an explicit set of groups, set them.
- Otherwise, if we're changing UID and the old UID was 0, call setgroups with `ptr::null()` to drop our groups. (And update the comment accordingly.)
- Otherwise (we weren't given a set of groups and we're not changing UID from 0), do nothing.